services/121-service/src/registration/services/bulk-import.service.ts

```diff
@@ -380,4 +389,100 @@ export class BulkImportService {
     }
     return validatatedArray;
   }
+
+  public async importFspReconciliation(
+    csvFile,
+    programId: number,
+    userId: number,
+  ): Promise<ImportResult> {
+    const validatedImportRecords = await this.csvToValidatedFspReconciliation(
+      csvFile,
+    );
+
+    let countImported = 0;
+    let countNotFound = 0;
+
+    const importResponseRecords = [];
+    for await (const record of validatedImportRecords) {
+      const importResponseRecord = record as UploadFspReconciliationResult;
+
+      const registration = await this.registrationRepository.findOne({
+        where: { referenceId: record.referenceId },
+        relations: ['fsp'],
+      });
+      if (!registration) {
+        importResponseRecord.importStatus = ImportStatus.unmatched;
+        importResponseRecords.push(importResponseRecord);
+        countNotFound += 1;
+        continue;
+      }
+
+      const paTransactionResult = new PaTransactionResultDto();
+      paTransactionResult.referenceId = record.referenceId;
+      paTransactionResult.status = record.status as StatusEnum;
+      paTransactionResult.fspName = registration.fsp.fsp as FspName;
+      paTransactionResult.message = '';
+      paTransactionResult.calculatedAmount = Number(record.amount);
+
+      await this.transactionsService.storeTransaction(
+        paTransactionResult,
+        programId,
+        Number(record.payment),
+      );
+
+      importResponseRecord.importStatus = ImportStatus.imported;
+      importResponseRecords.push(importResponseRecord);
+      countImported += 1;
+    }
+
+    this.actionService.saveAction(
+      userId,
+      programId,
+      AdditionalActionType.importPeopleAffected,
```

Comment:
`AdditionalActionType.importPeopleAffected` should be a new value?

Reply:
yes.

Reply:
done